node/node.go

```diff
@@ -970,12 +970,41 @@ func (node *AlgorandFullNode) loadParticipationKeys() error {
 			}
 		} else {
 			// Tell the AccountManager about the Participation (dupes don't matter)
+			// make sure that all stateproof data (with are not the keys per round)
+			// are being store to the registry in that point
 			added := node.accountManager.AddParticipation(part)
 			if added {
 				node.log.Infof("Loaded participation keys from storage: %s %s", part.Address(), info.Name())
 			} else {
 				part.Close()
+				continue
 			}
+			// create a const value Number of records in one fetch
+			// insert into the registry X rounds of keys via account manager
+			// all in for loop until no more keys in db
+			partID := part.ID()
+			numKeys := part.Participation.StateProofSecrets.CountKeys(part.Store)
+			for i := uint64(0); i < uint64(numKeys); i++ {
```

Comment:
what if this loop breaks?(ie node crashes)
can we recover from it?
do we have a test?

Reply:
Maybe I should just return error instead since I'm counting keys before starting to iterate, so the reason for this `break` not longer exists.